membership_product_set/models/res_partner.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2019 Yu Weng <yweng@elegosoft.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.multi
+    def create_membership_invoice(self, product_id=None, datas=None):
+        product_id = product_id or datas.get('membership_product_id')
+        amount = datas.get('amount', 0.0)
+        product = self.env['product.product'].browse(product_id)
+        invoice_list = []
+        if product.membership_set:
```

Comment:
The original [create_membership_invoice](https://github.com/OCA/OCB/blob/11.0/addons/membership/models/partner.py#L173) has a lot of code in common with this one ?

Try not to copy and paste it just for a certain condition. Try to make a method in the base method that perform the creation of **line_values** and just overwrite that method in this class.

My other suggestions are inherited from the base code being extended, but this is very important for the future maintainability of this method.

Reply:
My extension will only be executed, if product.membership_set is True. Otherwise the origin function will be called. This function can also be inherited by other module. I don't see any problem now. I don't want to change the origin function from Odoo Repo and create a new PR of it.

Reply:
ok